doc/Programming-guide/Key-platform-features/Configuration-management/index.md

```diff
@@ -58,104 +70,115 @@ The following examples illustrate basic constituents of the configuration schema
-    "type": "record",
-    "fields": [
+    "name":"rootT",
+    "namespace":"org.kaaproject.kaa.schema.sample",
+    "type":"record",
+    "fields":[
         ...
     ]
 }
 ```
 
 * The **name** and **namespace** attributes are both mandatory for the record type. They are used for the record referencing in derivative schemas.
 
-* The **optional** field attribute (boolean, false by default) determines whether or not the field in the record is optional. Internally, Kaa translates optional fields into union fields with the null type at the top of the list (which automatically makes them default to null - see the [Records auto-generation]() section for more details).
-In case of the optional union field, Kaa automatically puts null at the top of the types list in the union definition.
+* The **optional** field attribute (```boolean```, false by default) determines whether or not the field in the record is optional. Internally, Kaa translates
+optional fields into union fields with the ```null``` type at the top of the list (which automatically makes them default to ```null``` - see the
+[Records auto-generation](#records-auto-generation) section for more details). <br/>
+In case of the _optional_ union field, Kaa automatically puts ```null``` at the top of the types list in the union definition.
 
 ```json
 {
-    "name": "rootT",
-    "namespace": "org.kaaproject.sample",
-    "type": "record",
-    "fields": [
+    "name":"rootT",
+    "namespace":"org.kaaproject.kaa.schema.sample",
+    "type":"record",
+    "fields":[
         {
-            "name": "optionalBytesField",
-            "type": "bytes",
-            "optional": true
+            "name":"optionalBytesField",
+            "type":"bytes",
+            "optional":true
         },
         {
-            "name": "optionalSuit",
-            "optional": true,
-            "type": {
-                "name": "suitT",
-                "namespace": "org.kaaproject.sample",
-                "type": "enum",
-                "symbols" : ["spades", "hearts", "diamonds", "clubs"]
+            "name":"optionalSuit",
+            "optional":true,
+            "type":{
+                "name":"suitT",
+                "namespace":"org.kaaproject.kaa.schema.sample",
+                "type":"enum",
+                "symbols":[
+                    "spades",
+                    "hearts",
+                    "diamonds",
+                    "clubs"
+                ]
             }
         }
     ]
 }
 ```
 
-* The **by_default** field attribute (the value is interpreted according to the field type; it has no default value) determines the default value for the field and is used for generation of the default record. The **by_default** attribute must be present for all mandatory primitive record fields, except for null.
+* The **by_default** field attribute (the value is interpreted according to the field type; it has no default value) determines the default value for the field
```

Comment:
Please verify this, i believe that by_default is optional even for mandatory fields.